### ISSUE TITLE
Releases should show tag metadata when not overridden

### ISSUE THREAD

**Issue by clarfonthey** [go-gitea/gitea#4206]
- Gitea version (or commit ref): 1.4.2
- Git version: 2.17.1
- Operating system: Arch Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
  - [x] Not relevant
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant

## Description

Example: (dead link)

If you go and actually clone the repo, you'll find that the releases have separate messages from the commits they're attached to. It'd be nice if gitea could pull this data from git directly and show it as a release, so that it doesn't have to be stored in the database.

**Comment by 0x5c** [comment]
Yes, and/or tag annotations. It took me by surprise that Gitea didn't show the tag message as the release message. Especially since Gitea does not show releases separately from tags.

**Comment by clarfonthey** [comment]
Honestly, thinking about it more, release data really shouldn't be kept separate from git, minus the attachments. The tag should have all the release metadata.

**Comment by 0x5c** [comment]
Indeed, just like Github currently does:
![image](https://user-images.githubusercontent.com/5877043/54465203-59240780-4750-11e9-8420-38f8ab7ba214.png)
![image](https://user-images.githubusercontent.com/5877043/54465226-7f49a780-4750-11e9-84df-f0086dbaa5f8.png)

**Comment by clarfonthey** [comment]
GitHub still doesn't handle this, actually; the release message is still separate from the tag message. IMHO they should be the same.

**Comment by 0x5c** [comment]
Github seems to take the tag message as release note by default.  
*Unless* I missed something. Let me check real-quick.

**Comment by 0x5c** [comment]
Thinking about it (didn't have time to check yet), tags have other uses than *only* for releases. Maybe tags and releases should be separate (like on GitHub), but with a single button to make a release/note from a tag?

**Comment by clarfonthey** [comment]
IMHO releases should be a strict subset of tags. i.e., you can have tags that aren't releases, but every release is a tag. Which tags are releases can be stored in the DB, but the release message should always be exactly the same as the tag message.

**Comment by 0x5c** [comment]
Indeed  
And if considering releases as a subset of tags is too much and *non-acceptable* (ie: opinions), at-least have the release note be, by default, the tag message.

**Comment by tfardet** [comment]
If there is a desire to be able to keep tags and releases separate in some cases, would it be possible to:

- give the user tha ability to define a regex so that e.g. all vX.Y.Z tags are automatically considered releases
- make the tags content the default release message?

It is a bit tedious to have to create releases from tags every time, then renter the text.
